### PR TITLE
Add ratelimit for phone number verification

### PR DESCRIPTION
**Pull request by Konstantinov-Innokentii**
# What this PR does

## Checklist

- [x] Tests updated
- [x] `CHANGELOG.md` updated

### CODE REVIEW

File: engine/apps/api/tests/test_user.py [hunk 1]
```diff
@@ -1499,3 +1500,41 @@ def test_check_availability_other_user(make_organization_and_user_with_plugin_to
...
 
     assert response.status_code == status.HTTP_200_OK
+
+
+@patch("apps.twilioapp.phone_manager.PhoneManager.send_verification_code", return_value=Mock())
+@patch("apps.twilioapp.phone_manager.PhoneManager.verify_phone_number", return_value=(True, None))
+@patch("apps.api.apps.api.throttlers.GetPhoneVerificationCodeThrottler.get_throttle_limits ", return_value=(1, 10 * 60))
+@patch("apps.api.apps.api.throttlers.VerifyPhoneNumberThrottler.get_throttle_limits ", return_value=(1, 10 * 60))
+@pytest.mark.django_db
+def test_phone_number_verification_flow_rateliomit(
```
Comment (joeyorlando) [inline engine/apps/api/tests/test_user.py:1510]:
```suggestion
def test_phone_number_verification_flow_ratelimit(
```

File: engine/apps/public_api/throttlers/user_throttle.py [hunk 2]
```diff
@@ -2,42 +2,5 @@
...
-        self.history = self.cache.get(self.key, [])
-        self.now = self.timer()
-
-        # Drop any requests from the history which have now passed the
-        # throttle duration
-        while self.history and self.history[-1] <= self.now - self.duration:
-            self.history.pop()
-        if len(self.history) >= self.num_requests:
-            return self.throttle_failure()
-        return self.throttle_success()
```
Comment (joeyorlando) [inline engine/apps/public_api/throttlers/user_throttle.py:43]: I see most of this logic was moved to `CustomRateScopedThrottler`. Was the intent to have the `UserThrottle` class subclass `CustomRateScopedThrottler`?

File: engine/apps/api/views/user.py [hunk 3]
```diff
@@ -275,11 +276,14 @@ def current(self, request):
         serializer = UserSerializer(self.get_queryset().get(pk=self.request.user.pk))
         return Response(serializer.data)
 
-    @action(detail=False, methods=["get"])
+    @action(
+        detail=False,
+        methods=["get"],
+    )
```
Comment (joeyorlando) [inline engine/apps/api/views/user.py:282]:
```suggestion
    @action(detail=False, methods=["get"])
```

File: engine/apps/public_api/throttlers/user_throttle.py (hunk 2 above)
Comment (Konstantinov-Innokentii) [reply engine/apps/public_api/throttlers/user_throttle.py:43]: UserThrottle no more requires set rate per custom period

Comment (Konstantinov-Innokentii) [reply engine/apps/public_api/throttlers/user_throttle.py:43]: Actually it should be called PublicAPIThrottler

File: engine/apps/api/tests/test_user.py [hunk 4]
```diff
@@ -1499,3 +1505,88 @@ def test_check_availability_other_user(make_organization_and_user_with_plugin_to
...
+@patch("apps.twilioapp.phone_manager.PhoneManager.verify_phone_number", return_value=(True, None))
+@patch(
+    "apps.api.throttlers.GetPhoneVerificationCodeThrottlerPerUser.get_throttle_limits ",
+    return_value=(1, 10 * 60),
+)
+@patch("apps.api.throttlers.VerifyPhoneNumberThrottlerPerUser.get_throttle_limits ", return_value=(1, 10 * 60))
+@pytest.mark.django_db
+def test_phone_number_verification_flow_ratelimit_per_user(
+    moch_get_phone_verify_phone_number_limits,
+    moch_get_phone_verification_code_get_throttle_limits,
```
Comment (joeyorlando) [inline engine/apps/api/tests/test_user.py:1520]:
```suggestion
    mock_get_phone_verify_phone_number_limits,
    mock_get_phone_verification_code_get_throttle_limits,
```

File: engine/apps/api/tests/test_user.py [hunk 5]
```diff
@@ -1499,3 +1505,88 @@ def test_check_availability_other_user(make_organization_and_user_with_plugin_to
...
+@patch("apps.twilioapp.phone_manager.PhoneManager.verify_phone_number", return_value=(True, None))
+@patch(
+    "apps.api.throttlers.GetPhoneVerificationCodeThrottlerPerOrg.get_throttle_limits ",
+    return_value=(1, 10 * 60),
+)
+@patch("apps.api.throttlers.VerifyPhoneNumberThrottlerPerOrg.get_throttle_limits ", return_value=(1, 10 * 60))
+@pytest.mark.django_db
+def test_phone_number_verification_flow_ratelimit_per_org(
+    moch_get_phone_verify_phone_number_limits,
+    moch_get_phone_verification_code_get_throttle_limits,
```
Comment (joeyorlando) [inline engine/apps/api/tests/test_user.py:1562]:
```suggestion
    mock_get_phone_verify_phone_number_limits,
    mock_get_phone_verification_code_get_throttle_limits,
```

**Review by joeyorlando — APPROVED** [review]
LGTM, just some minor nit comments

**Pull request merged**